test-validator/src/lib.rs

```diff
@@ -312,6 +314,38 @@ impl TestValidatorGenesis {
         Ok(self)
     }
 
+    pub fn clone_upgradeable_programs<T>(
+        &mut self,
+        addresses: T,
+        rpc_client: &RpcClient,
+    ) -> Result<&mut Self, String>
+    where
+        T: IntoIterator<Item = Pubkey> + Clone,
```

Comment:
We could avoid the clone bound by making `clone_accounts` take an `Iterator<Item = &Pubkey>` and then in the first line of `clone_accounts` do
```
let addresses: Vec<Pubkey> = addresses.cloned().collect();
```
Then, whenever we call it, do it with an `iter()`, ie `self.clone_accounts(addresses.iter(), ...)`. We'd also have to consume the `IntoIterator` at the start of this function and collect it into a vec to make this work.

All that to say, the `Clone` bound is a bit annoying, but it makes things more legible, so I'm happy to stick with it.

Reply:
we can also remove the bound by just collecting the addresses into vec

Reply:
That's true. Either way, not a big deal :smile: